django/db/models/constraints.py

```diff
@@ -125,8 +139,9 @@ def remove_sql(self, model, schema_editor):
 
     def validate(self, model, instance, exclude=None, using=DEFAULT_DB_ALIAS):
         against = instance._get_field_value_map(meta=model._meta, exclude=exclude)
+        check = get_check_q(self.check, model)
```

Comment:
Is it not enough to add `pk` to the field map (`_get_field_value_map`)?

Reply:
@felixxm sure, much more cleaner, thanks